ideas/143-seamless-login.md

```diff
@@ -0,0 +1,90 @@
+## Preamble
+
+    Idea: 143
+    Title: Seamless Login
+    Status: Draft
+    Created: 2018-05-04
+
+
+## Summary
+
+When a user opens Status or taps on a push notification they should be able to start using the app securely, quickly and without any interruptions.
+
+## Swarm Participants
+- Lead Contributor: @alwx
+- Testing & Evaluation: @asemiankevich
+- PM: @chadyj
+- UX: @andmironov
+- UXR: @jpbowen @hesterbruikman
+
+
+## Product Overview
+
+Status frequently logs out users when the app is in the background and forces users to log in again when opening the app. Being repeatedly and unexpectedly logged out has led to frustration and complaints (see Instabug and internal reports) which negatively impact the user experience and potentially contributes to churn. Also, users are encouraged to have complex passwords but this can be cumbersome to enter.
+
+By reducing friction around the critical point of returning to the app users can spend more time in Status which will help DAU and retention (Q2 OKR).
+
+This idea seeks to make logging in seamless through both technical and UX methods.
+
+
+### Product Description
+
+The Status login process should smooth as possible, and this can be achieved with the (optional) use of device security features such as Touch ID or a pin code. In addition to the password prompt, a user can use choose to securely use their fingerprint. This will be set up during onboarding and toggled in settings.
+
+Note, all transactions require authentication, so even if the phone was lost or stolen a third party couldn’t transfer funds. This idea does not cover the transaction flow.
+
+For future iterations, this swarm will research the security implications and explore enhancements around logging in, logging out, and recovering from crashes.
+
+### Security and Privacy Implications
+
+Potentially, both Touch ID ("fingerprint authentication") and Face ID can improve security by allowing users to have stronger passwords (the current experience encourages to have a weak password). Of course, it has some disadvantages, but we're going to require passwords anyway for important operations, so the goal is to improve security anyway.
+Moreover, options like "Require authentication every time I open the application" will improve both security and privacy for sure – right now we ask password only in the case user was accidentally logged out.  
+
+Of course, there are some open questions:
+1) We should discuss options like requiring passwords to log in once in a while even if biometrical authentication is used;
+2) Screens like "Settings" or "Backup your Seed Phrase" should be protected by password anyway;
+3) We need to discuss the "level of freedom" – what should be the options? 
+4) How can we encrypt/decrypt Realm in the situation when we don't have a password?
```

Comment:
What does it mean exactly? 

As far I understand we don't need a password to encrypt realm, just a key which is in the keychain and can be randomly generate ( which I take can be unlocked using touch id/face id).

Reply:
You're right, let's remove this one.